service/history/history_engine.go

```diff
@@ -819,10 +820,6 @@ func (e *historyEngineImpl) NotifyNewTasks(
 	}
 }
 
-func (e *historyEngineImpl) AddSpeculativeWorkflowTaskTimeoutTask(task *tasks.WorkflowTaskTimeoutTask) {
-	e.queueProcessors[tasks.CategoryMemoryTimer].NotifyNewTasks([]tasks.Task{task})
-}
-
```

Comment:
Not sure why this is needed exactly? `NotifyNewTasks` seems to work the same way.

Reply:
There was a reason why I added this one liner. But I don't remember it and ok to remove.